include/LightGBM/boosting.h

```diff
@@ -28,12 +28,12 @@ class Boosting {
   * \param train_data Training data
   * \param object_function Training objective function
   * \param training_metrics Training metric
-  * \param output_model_filename Filename of output model
   */
-  virtual void Init(const Dataset* train_data,
+  virtual void Init(
+    const BoostingConfig* config,
+    const Dataset* train_data,
     const ObjectiveFunction* object_function,
-    const std::vector<const Metric*>& training_metrics,
-    const char* output_model_filename) = 0;
+    const std::vector<const Metric*>& training_metrics) = 0;
```

Comment:
do you change the training_metrics in Init function? I not sure using "&" here is perfect...